numpy/_core/src/umath/string_buffer.h

```diff
@@ -1505,4 +1505,81 @@ string_expandtabs(Buffer<enc> buf, npy_int64 tabsize, Buffer<enc> out)
 }
 
 
+enum class JUSTPOSITION {
+    CENTER, LEFT, RIGHT
+};
+
+template <ENCODING enc>
+static inline npy_intp
+string_pad(Buffer<enc> buf, npy_int64 width, Buffer<enc> fill, JUSTPOSITION pos, Buffer<enc> out)
+{
+    size_t finalwidth = width > 0 ? width : 0;
+    size_t len = buf.num_codepoints();
+    if (len >= finalwidth) {
+        buf.buffer_memcpy(out, len);
+        out.buffer_fill_with_zeros_after_index(len);
+        return (npy_intp) len;
+    }
+
+    size_t left, right;
+    if (pos == JUSTPOSITION::CENTER) {
+        size_t pad = finalwidth - len;
+        left = pad / 2 + (pad & finalwidth & 1);
```

Comment:
Why the `& finalwidth`? I'd have thought `pad & 1` would do (or indeed `left = (pad + 1) / 2`)

EDIT: well, it looks like python itself does what you have, so ignore this!

Reply:
Yes, I was a bit confused by this as well, but decided to just take what CPython does.